src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
 public class FileWork {
+    private static final String SPECIFIED_CHARACTER = "w";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            List<String> stringList = Files.readAllLines(file.toPath());
+            if (stringList.size() == 0) {
+                return new String[0];
+            }
+            for (String string: stringList) {
+                String[] words = string.split(" ");
+                for (String word : words) {
+                    if (word.toLowerCase().startsWith(SPECIFIED_CHARACTER)) {
+                        stringBuilder.append(word.toLowerCase().replaceAll("[^a-z]", ""))
+                                .append(" ");
+                    }
+                }
+            }
+            if (stringBuilder.toString().isEmpty()) {
+                return new String[0];
+            }
+            String[] sortableStrings = stringBuilder.toString().split(" ");
+            ArrayList<String> sortingList = new ArrayList<>(Arrays.asList(sortableStrings));
```

Comment:
```suggestion
           List<String> sortingList = Arrays.asList(sortableStrings);
```
Or you want to create a deep copy?